front/src/reducers/editor/index.ts

```diff
@@ -38,6 +46,18 @@ export const editorSlice = createSlice({
   initialState: editorInitialState,
   reducers: {
     ...buildInfraStateReducers<EditorState>(),
+    updateEditorViewportAction: (state, action: PayloadAction<Partial<Viewport>>) => {
+      state.mapSettings.viewport = { ...state.mapSettings.viewport, ...action.payload };
+    },
+    updateMapSettings: (state, action: PayloadAction<Partial<MapSettings>>) => {
+      state.mapSettings = {
+        ...state.mapSettings,
+        ...action.payload,
+      };
+    },
```

Comment:
You can use `...buildMapStateReducer<OsrdStdcmConfState>()` I think
```suggestion
    ...buildMapStateReducer<OsrdStdcmConfState>(),
```